static/app/views/settings/projectGeneralSettings/index.tsx

```diff
@@ -250,182 +259,165 @@ class ProjectGeneralSettings extends DeprecatedAsyncComponent<Props, State> {
+      setApiQueryData(queryClient, makeProjectSettingsQueryKey, resp);
+      if (projectId !== resp.slug) {
+        changeProjectSlug(projectId, resp.slug);
+        // Container will redirect after stores get updated with new slug
+        onChangeSlug(resp.slug);
+      }
+      // This will update our project context
+      ProjectsStore.onUpdateSuccess(resp);
+    },
+  };
 
-class ProjectGeneralSettingsContainer extends Component<ContainerProps> {
-  componentWillUnmount() {
-    this.unsubscribe();
-  }
+  const projectIdField: FieldObject = {
+    name: 'projectId',
+    type: 'string',
+    disabled: true,
+    label: t('Project ID'),
+    setValue(_, _name) {
+      return project.id;
+    },
+    help: t('The unique identifier for this project. It cannot be modified.'),
+  };
 
-  changedSlug: string | undefined = undefined;
-  unsubscribe = ProjectsStore.listen(() => this.onProjectsUpdate(), undefined);
+  return (
+    <div>
+      <SentryDocumentTitle title={t('Project Settings')} projectSlug={project.slug} />
+      <SettingsPageHeader title={t('Project Settings')} />
+      <ProjectPermissionAlert project={project} />
+      <Form {...formProps}>
+        <JsonForm
+          {...jsonFormProps}
+          title={t('Project Details')}
+          fields={[fields.name, projectIdField, fields.platform]}
+        />
+        <JsonForm {...jsonFormProps} title={t('Email')} fields={[fields.subjectPrefix]} />
+      </Form>
+      <Hook name="spend-visibility:spike-protection-project-settings" project={project} />
+      <Form {...formProps}>
+        <JsonForm
+          {...jsonFormProps}
+          title={t('Event Settings')}
+          fields={[fields.resolveAge]}
+        />
 
-  onProjectsUpdate() {
-    if (!this.changedSlug) {
-      return;
-    }
-    const project = ProjectsStore.getBySlug(this.changedSlug);
+        <JsonForm
+          {...jsonFormProps}
+          title={t('Client Security')}
+          fields={[
+            fields.allowedDomains,
+            fields.scrapeJavaScript,
+            fields.securityToken,
+            fields.securityTokenHeader,
+            fields.verifySSL,
+          ]}
+          renderHeader={() => (
+            <PanelAlert type="info">
+              <TextBlock noMargin>
+                {tct(
+                  'Configure origin URLs which Sentry should accept events from. This is used for communication with clients like [link].',
+                  {
+                    link: (
+                      <ExternalLink href="https://github.com/getsentry/sentry-javascript">
+                        sentry-javascript
+                      </ExternalLink>
+                    ),
+                  }
+                )}{' '}
+                {tct(
+                  'This will restrict requests based on the [Origin] and [Referer] headers.',
+                  {
+                    Origin: <code>Origin</code>,
+                    Referer: <code>Referer</code>,
+                  }
```

Comment:
```suggestion
                  'This will restrict requests based on the [code:Origin] and [code:Referer] headers.',
                  {
                    code: <code />,
                  }
```